src/main/java/seedu/address/storage/JsonAdaptedTask.java

```diff
@@ -0,0 +1,93 @@
+package seedu.address.storage;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.model.contact.Contact;
+import seedu.address.model.tag.Tag;
+import seedu.address.model.task.Date;
+import seedu.address.model.task.Priority;
+import seedu.address.model.task.Status;
+import seedu.address.model.task.Task;
+import seedu.address.model.task.TaskName;
+
+/**
+ * Jackson-friendly version of {@link Contact}.
+ */
+public class JsonAdaptedTask {
+
+    public static final String MISSING_FIELD_MESSAGE_FORMAT = "Task's %s field is missing!";
+
+    private final String name;
+    private final JsonAdaptedTag tag;
+    private final String priority;
+    private final String date;
+    private final String status;
+
+    /**
+     * Constructs a {@code JsonAdaptedContact} with the given contact details.
+     */
+    @JsonCreator
+    public JsonAdaptedTask(@JsonProperty("name") String name, @JsonProperty("tag") JsonAdaptedTag tag,
+                              @JsonProperty("priority") String priority,
+                           @JsonProperty("date") String date, @JsonProperty("status") String status) {
+        this.name = name;
+        this.tag = tag;
+        this.priority = priority;
+        this.date = date;
+        this.status = status;
+    }
+
+    /**
+     * Converts a given {@code Task} into this class for Jackson use.
+     */
+    public JsonAdaptedTask(Task source) {
```

Comment:
I think the implementation of this constructor can be improved. For a Task object, the Date, Priority and Tag fields are optional and the below implementation might result in a NullPointerException, for eg, invoking getPriority().get() might return null.